app/src/androidTest/java/com/blockeq/stellarwallet/WalletRecoveryTest.kt

```diff
@@ -0,0 +1,92 @@
+package com.blockeq.stellarwallet
+
+import android.support.test.rule.ActivityTestRule
+import android.support.test.runner.AndroidJUnit4
+import com.blockeq.stellarwallet.activities.LaunchActivity
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+/**
+ * Instrumented test, which will execute on an Android device.
+ *
+ * IMPORTANT: make sure that the app and previous tests are not installed in the device.
+ * Run sh uninstallApk.sh first. This could be solved using orchestrator tests.
+ *
+ * See [testing documentation](http://d.android.com/tools/testing).
+ */
+
+@RunWith(AndroidJUnit4::class)
+class WalletRecoveryTest {
```

Comment:
In general the test are good but you are not verifying if the recovered account is the expected wallet (secret seed or account id) during mnemonic recovery and mnemonic24 + passphrase.

Another good test is to check the if the mnemonic produced under settings is the one expected.
Same for passphrase.

Reply:
Okay that's true, I'll add checks for the account id